optimum/bettertransformer/models/decoder_models.py

```diff
@@ -74,12 +76,15 @@ def wrapped_scaled_dot_product(
                     torch.bool
                 )
 
-                causal_mask = torch.where(causal_mask, 0, self._mask_value)
+                causal_mask = torch.where(causal_mask, 0, self._mask_value).to(value.dtype)
```

Comment:
This cast is bad. Can we instead move the definition of `mask_value` in the forward?